bin/samples_checker.py

```diff
@@ -101,24 +101,25 @@ def read_metadata_json(json_file):
         )
 
 
-def resolve_vcf_file_location(vcf_dir_prefix, files_per_analysis):
-    result_files_per_analysis = {}
-    for analysis_alias in files_per_analysis:
-        result_files_per_analysis[analysis_alias] = []
-        for f in files_per_analysis[analysis_alias]:
-            if vcf_dir_prefix:
-                # Remove leading / to ensure Join concatenate the path
-                if os.path.isabs(f):
-                    file_path = os.path.join(vcf_dir_prefix, *pathlib.Path(f).parts[1:])
-                else:
-                    file_path = os.path.join(vcf_dir_prefix, f)
-            else:
-                file_path = os.path.abspath(f)
-            print(f, vcf_dir_prefix, file_path)
-            if os.path.exists(file_path):
-                result_files_per_analysis[analysis_alias].append(file_path)
-            else:
-                raise FileNotFoundError(f'{file_path} cannot be resolved')
+def resolve_vcf_file_location(vcf_files, files_per_analysis):
```

Comment:
I found this method confusing, maybe could use a docstring... I guess it's checking concordance between the VCF files provided in the metadata (`files_per_analysis`) and the ones passed on the command-line (`vcf_files`), the latter of which gives the full path... if this is true it feels like the resolving of location is kind of incidental now.

As an aside, I think I understand better the complexity you were talking about, it kind of feels like we should _only_ receive metadata on the command line (including the files/analysis association) and use that as our sole source of truth. Then the metadata is more of a self-contained, complete description of the project, and there's less room for error (both user error and programmer error).